### PR TITLE
Encode date range second

### PR DESCRIPTION
**Pull request by rasaigao**
## Description / JIRA Links
Item         | Description
------------ | -------------
[SERVINFRA-1236](https://tophat.atlassian.net/browse/SERVINFRA-1236) | Title

- [ ] I have updated the changelog.

### Tests
- Insufficient tests.

### Code Review
- This PR addresses comments from a previous PR (insert link to previous PR)

### CODE REVIEW

File: src/linkedin_profile_data/date_manager.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "Oct": 10,
+    "Nov": 11,
+    "Dec": 12
+}
+
+
+class DateRangeManager:
+
+    @staticmethod
+    def parse_date_range(dr):
```
Comment (owenchak) [inline src/linkedin_profile_data/date_manager.py]: A more descriptive name for `dr` would be good here

File: src/linkedin_profile_data/date_manager.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        st_date, en_date = dr.split(" – ")
```
Comment (owenchak) [inline src/linkedin_profile_data/date_manager.py]: could just use `start_date` and `end_date`

File: src/linkedin_profile_data/date_manager.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        end_formatted = DateRangeManager._format_date(en_date)
+        date_range_formatted = [start_formatted, end_formatted]
+        return date_range_formatted
+
+    @staticmethod
+    def _format_date(date_string: str) -> str:
+        epoch = datetime.datetime.utcfromtimestamp(0).date()
+        if "Present" in date_string:
+            date = datetime.datetime.now().date()
+
```
Comment (owenchak) [inline src/linkedin_profile_data/date_manager.py]: Could you add some comments to explain each edge case please

File: src/linkedin_profile_data/views.py [hunk 4]
```diff
@@ -60,61 +61,6 @@
     },
 ]
 
-# TODO: This is to be removed
-sample_company_employees_res = [
```
Comment (owenchak) [inline src/linkedin_profile_data/views.py]: Why did you remove this :(

File: src/linkedin_profile_data/date_manager.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+
+class DateRangeManager:
+
+    @staticmethod
+    def parse_date_range(dr):
+        st_date, en_date = dr.split(" – ")
+        start_formatted = DateRangeManager._format_date(st_date)
+        end_formatted = DateRangeManager._format_date(en_date)
```
Comment (owenchak) [inline src/linkedin_profile_data/date_manager.py]: You could make this a static method and just do `self._format_date(en_date)`

File: src/linkedin_profile_data/views.py [hunk 6]
```diff
@@ -202,11 +148,16 @@ def _extract_work_history(profile_data: dict) -> list:
...
-            #  easier to create the cross validation algorithm
             date_range = job.get("dateRange")
+            date_range_formatted = DateRangeManager.parse_date_range(date_range)
             res.append(
-                {"company_name": company_name, "date_range": date_range}
+                {
+                    "company_name": company_name,
+                    "dates": {
+                        "start_date": date_range_formatted[0],
+                        "end_date": date_range_formatted[1]
```
Comment (owenchak) [inline src/linkedin_profile_data/views.py]: Could you remove "dates" from the API response body please. My initial intention was displaying the date_range a candidate has worked with a current Top Hat employee but this seems like a useless feature lol

**Comment by owenchak** [comment]
Do you have steps to devqa this?

File: src/linkedin_profile_data/views.py (hunk 6 above)
Comment (rasaigao) [reply src/linkedin_profile_data/views.py]: Just talked about it :)
